src/v3/src/components/Widget/index.tsx

```diff
@@ -438,23 +439,33 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
     if (isClientTransaction) {
       return;
     }
-    if (widgetRendered && typeof idxTransaction !== 'undefined'
-      && uischema.elements.length > 0) {
-      eventEmitter?.emit?.('afterRender', getEventContext(idxTransaction));
+    if (widgetRendered) {
+      (async () => {
+        const isFirstRender = typeof widgetRenderedOnce.current === 'undefined';
+        const executeAfterHooks = () => widgetHooks && widgetHooks.callHooks('after', idxTransaction, responseError);
```

Comment:
Optional chaining simplifies this statement:
```suggestion
        const executeAfterHooks = () => widgetHooks?.callHooks('after', idxTransaction, responseError);
```